File: 3-strings/1243.py
def isAWord(string):
    if string[0]=='.':return 0
    dif=0
    for i in string:
        if i=='.' and string.index('.')==len(string)-1:dif=1;break
        if (ord(i)<65 or ord(i)>90) and (ord(i)<97 or ord(i)>122):return 0
    return len(string)-dif
while 1:
    try:
        totStrings,average=0,0
        for _ in list(map(isAWord,input().split())):
            average+=_
            totStrings+=1 if _ else 0
        if not totStrings or not average:print('250');continue
        result=average//totStrings
        print('250')if result<4 else print('500') if result<6 else print('1000')
    except EOFError:break

File: 3-strings/1243.cpp
#include<iostream>
#include<cstring>
//
int isAWord(char*string){
	if((int)string[0]==46)return 0;
	int i;
	for(i=0;string[i]!='\n'&&string[i]!='\0';++i){
		if((int)string[i]==46&&(string[i+1]=='\n'|string[i+1]=='\0'))break;
		else if(((int)string[i]<65|(int)string[i]>90)&&((int)string[i]<97|(int)string[i]>122))return 0;
	}
	int length=0;
	for(int j=0;string[j]!='\n'&&string[j]!='\0'&&string[j]!='.';++j){
		++length;
	}
	return length;
}
//
int main(){
	std::string inp;
	while(std::getline(std::cin,inp)){
		if(!std::cin)break;
		char*split=std::strtok(&inp[0]," ");
		int average=0,tots=0;
		while(split!=NULL){
			int temp=isAWord(split);
			if(temp){
				average+=temp;
				++tots;
			}
			split=std::strtok(NULL," ");
		}
		(!average)?average=1:average;
		(!tots)?tots=1:tots;
		int result=int((double)average/(double)tots);
		//std::cout<<result<<" "<<average<<"\n";
		if(result<4)std::cout<<"250\n";
		else if(result<6)std::cout<<"500\n";
		else std::cout<<"1000\n";
	}
	return 0;
}
